src/sql/parts/notebook/notebook.component.ts

```diff
@@ -526,6 +424,15 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 		provider.bindTo(this.contextKeyService);
 	}
 
+	public async save(): Promise<boolean> {
+		try {
+			return this._notebookParams.input.save();
+		} catch (err) {
```

Comment:
I'd suggest don't handle rejection, as the core save implementation does most of this. The caller should expect potential for rejected promise and be able to decide what to do itself.